### ISSUE TITLE
xdg-desktop-portal backend on Linux returns URI encoded paths

### ISSUE THREAD

**Issue by WerWolv** [btzy/nativefiledialog-extended#65]
Hi

I recently switched from the GTK backend to the Portals backend on Linux to allow file dialogs to work properly in Flatpaks and AppImages. Switching over was easy, however the xdg-desktop-portal backend returns all paths in URI encoding.

For example:
`/home/werwolv/Hello World.txt` becomes `/home/werwolv/Hello%20World.txt`

That behaviour is different from all the other backend which directly returns the normal path.

Is this an oversight in the library?
If it's the expected behaviour, what is the intended way to deal with these paths?

**Comment by btzy** [comment]
Thanks for the report.

I am able to reproduce this with the examples:
```
$ ./test_opendialog_c
Success!
/home/bernard/Documents/test%20folder/tmp.cpp
```

I checked the [Portal documentation](https://flatpak.github.io/xdg-desktop-portal/) and it seems that while it doesn't talk about URI encoding, it says the portal method returns a URI, which probably means that it should be URI-encoded.  This agrees with the discussion in https://github.com/flatpak/xdg-desktop-portal/issues/799, so it looks like a bug.

**Comment by btzy** [comment]
@WerWolv I've made a fix for this issue on #67.  Do you mind checking that it works for you?
